### PR TITLE
fix(ansi): update truncateLeft description, fix edge case

### PR DESCRIPTION
**Pull request by pranavkrishnan02**
Changes:

- Fixed `truncateLeft` description based on @caarlos0 's [PR](https://github.com/charmbracelet/x/pull/301)
- Fixed edge case: Return string if we're truncating 0 chars from the left (see test case) 

Issue: #329

### CODE REVIEW

File: ansi/truncate.go [hunk 1]
```diff
@@ -122,13 +122,13 @@ func Truncate(s string, length int, tail string) string {
 	return buf.String()
 }
 
-// TruncateLeft truncates a string from the left side to a given length, adding
-// a prefix to the beginning if the string is longer than the given length.
+// TruncateLeft truncates a string from the left side by a given number of characters, adding
+// a prefix to the beginning if the string is longer than the given number of characters.
 // This function is aware of ANSI escape codes and will not break them, and
 // accounts for wide-characters (such as East-Asian characters and emojis).
```
Comment (caarlos0) [inline ansi/truncate.go:128]: Reading this again, it still is a bit confusing I think.

Maybe something like:

```
TruncateLeft returns the last n characters [length] of the given string, 
adding a prefix to the beginning if the given string is longer than [length].
This function is aware of ANSI escape sequences and will not break them.
It also accounts for wide-characters (such as East-Asian characters and emojis).
```

I would also leave the variable named length I think, as to conform with `Truncate`

Comment (pranavkrishnan02) [reply ansi/truncate.go:128]: I think using `length` is confusing because the current implementations of `Truncate` and `TruncateLeft` aren't complimentary:

`Truncate(s string, length int, "")` returns the first `length` characters of s
`TruncateLeft(s string, length int, "")` returns the last `StringWidth(s) - length` characters of s

Making them complimentary would be ideal but it would break backward compatibility. It would probably look something like:

```
// Returns last [length] characters of s
func TruncateLeft(s string, length int, prefix string) string {
	charsToRemove := StringWidth(s) - length
	if charsToRemove <= 0 {
		return s
	}
        ...
}
```

Comment (caarlos0) [reply ansi/truncate.go:128]: right, truncate left was added to be used in `Cut`, mostly so we do things like horizontal scrolling. 


anyway its fine.

Comment (pranavkrishnan02) [reply ansi/truncate.go]: ok, tried to make it a bit more clear, should be good now

**Review by caarlos0 — APPROVED** [review]

**Review by aymanbagabas — APPROVED** [review]

**Comment by caarlos0** [comment]
Thanks @pranavkrishnan02 🙏
